File: src/Scripting/Ruby/patterns/microkernel.rb
# frozen_string_literal: true

module RubyPatterns
  module Microkernel
    module_function

    def run
      plugins = { upper: ->(text) { text.upcase } }
      raise 'Microkernel failed' unless plugins.fetch(:upper).call('plugin') == 'PLUGIN'
      true
    end
  end
end

RubyPatterns::Microkernel.run if $PROGRAM_NAME == __FILE__

File: src/Systems/C/patterns/microkernel.c
#include <stdbool.h>
static int a(int x){return x+1;}static int b(int x){return x*2;}bool run(void){int(*p[2])(int)={a,b};int v=3;for(int i=0;i<2;i++)v=p[i](v);return v==8;}
